src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -4674,6 +4687,121 @@ def _get_tools_cache_key(self, tools):
+                    "role": "user", 
+                    "content": prompt
+                })
+                if result.content:
+                    self._append_to_chat_history({
+                        "role": "assistant", 
+                        "content": result.content
+                    })
+            
+            return result.content if result else None
+            
+        except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)